include/openthread/platform/crypto.h

```diff
@@ -480,15 +569,81 @@ void otPlatCryptoRandomDeinit(void);
  *
- * @retval OT_ERROR_NONE     Successfully filled buffer with random values.
- * @retval OT_ERROR_FAILED   Operation failed.
+ * @retval OT_ERROR_NONE          Successfully filled buffer with random values.
+ * @retval OT_ERROR_FAILED        Operation failed.
  *
  */
 otError otPlatCryptoRandomGet(uint8_t *aBuffer, uint16_t aSize);
 
+/**
+ * Generate and populate the output buffer with a new ECDSA key-pair.
+ *
+ * @param[out] aKeyPair           A pointer to an ECDSA key-pair structure to store the generated key-pair.
+ *
+ * @retval OT_ERROR_NONE          A new key-pair was generated successfully.
+ * @retval OT_ERROR_NO_BUFS       Failed to allocate buffer for key generation.
+ * @retval OT_ERROR_NOT_CAPABLE   Feature not supported.
+ * @retval OT_ERROR_FAILED        Failed to generate key-pair.
+ *
+ */
+otError otPlatCryptoEcdsaGenerateKey(otPlatCryptoEcdsaKeyPair *aKeyPair);
+
+/**
+ * Get the associated public key from the input context.
+ *
+ * @param[in]  aKeyPair           A pointer to an ECDSA key-pair structure where the key-pair is stored.
+ * @param[out] aPublicKey         A pointer to an ECDSA public key structure to store the public key.
+ *
+ * @retval OT_ERROR_NONE          Public key was retrieved successfully, and @p aBuffer is updated.
+ * @retval OT_ERROR_PARSE         The key-pair DER format could not be parsed (invalid format).
+ * @retval OT_ERROR_INVALID_ARGS  The @p aContext is NULL.
+ *
+ */
+otError otPlatCryptoEcdsaGetPublicKey(const otPlatCryptoEcdsaKeyPair *aKeyPair, otPlatCryptoEcdsaPublicKey *aPublicKey);
+
+/**
+ * Calculate the ECDSA signature for a hashed message using the private key from the input context.
+ *
+ * This method uses the deterministic digital signature generation procedure from RFC 6979.
+ *
+ * @param[in]  aKeyPair           A pointer to an ECDSA key-pair structure where the key-pair is stored.
+ * @param[in]  aHash              A pointer to a SHA-256 hash structure where the hash value for signature calculation
+ * is stored.
+ * @param[out] aSignature         A pointer to an ECDSA signature structure to output the calculated signature.
+ *
+ * @retval OT_ERROR_NONE          The signature was calculated successfully, @p aSignature was updated.
+ * @retval OT_ERROR_PARSE         The key-pair DER format could not be parsed (invalid format).
+ * @retval OT_ERROR_NO_BUFS       Failed to allocate buffer for signature calculation.
+ * @retval OT_ERROR_INVALID_ARGS  The @p aContext is NULL.
+ *
+ */
+otError otPlatCryptoEcdsaSign(const otPlatCryptoEcdsaKeyPair *aKeyPair,
+                              const otPlatCryptoSha256Hash *  aHash,
+                              otPlatCryptoEcdsaSignature *    aSignature);
+
+/**
+ * Use the key from the input context to verify the ECDSA signature of a hashed message.
+ *
+ * @param[in] aPublicKey          A pointer to an ECDSA public key structure where the public key for signature
+ * verification is stored.
```

Comment:
We typically left-align the text wrap to the same column as below:
```suggestion
 * @param[in] aPublicKey          A pointer to an ECDSA public key structure where the public key for signature
 *                                verification is stored.
```